compliance_checker/cf/cf.py

```diff
@@ -132,7 +132,7 @@ def setup(self, ds):
 
         :param netCDF4.Dataset ds: An open netCDF dataset
         """
-        self._find_coord_vars(ds)
+        self.coord_vars = self._find_coord_vars(ds)
```

Comment:
Was the previous call `self._find_coord_vars` without the assignment assigning the coordinate variables internally?

Reply:
Looks like the function has a side effect and sets the `_coord_vars` variable in addition to returning the list of variables from the function.  The function is called several times elsewhere -- I'd prefer to avoid this side effect, and keep the object state changes in `setup` and then use `self.coord` vars instead of the function in aforementioned places.